### PR TITLE
Add option to specify the mime type for files matching a pattern

### PR DESCRIPTION
**Pull request by debbiedub**
The option is saved per added site.

First time that this version is used, the option is saved for all sites. This is not different from how other options are handled but could be confusing because on subsequent updates, the option is ignored and the value is set for all sites in your config-dir, even the ones you do not update.

### CODE REVIEW

File: freesitemgr [hunk 1]
```diff
@@ -271,6 +271,10 @@ def help():
     print("          - index file (default is index.html)")
     print("  -m, --mime-type")
     print("          - mime-type of the index file (default is \"text/html\")")
+    print("  --mime-type-match=PATTERN=MIME/TYPE")
+    print("     Set mime-type for files matching PATTERN. Can be given")
+    print("     multiple times to match distinct patterns.")
```
Comment (debbiedub) [inline freesitemgr:276]: There are a lot more to say about this but I don't think this is the place.

Among other things:
- If no files match, the mime-type will be determined my fcp3/node.py guestMimeType(filename) that in turn uses mimetypes.guess_type(filename, False). 
- The order is significant, i.e. `--mime-type-match=*.gif=image/gif --mime-type-match=databasefiles/*=text/plain` will, for the file `databasefiles/somedir/*.gif` give the mime-type `image/gif` while -`-mime-type-match=databasefiles/*=text/plain --mime-type-match=*.gif=image/gif` will give the mime-type `text/plain` for the same file name.

Comment (ArneBab) [reply freesitemgr:276]: Can you add that information in a file in doc/? Maybe with an example how to use this to upload a repo for dgof? Maybe as doc/freesitemgr.org ? (in org-mode format — markdown would also work)

**Review by ArneBab — APPROVED** [review]

File: freesitemgr (hunk 1 above)
Comment (debbiedub) [reply freesitemgr:276]: OK, yes. I will see if I understand the org-mode format. I will do that in another pull request.

**Comment by ArneBab** [comment]
Merged — thank you!

**Pull request merged**